app/common/src/main/java/com/fsck/k9/backends/AndroidAlarmManager.kt

```diff
@@ -62,12 +63,22 @@ class AndroidAlarmManager(
 
     override fun setAlarm(triggerTime: Long, callback: Callback) {
         this.callback.set(callback)
-        alarmManager.scheduleAlarm(triggerTime, pendingIntent)
+
+        if (AlarmManagerCompat.canScheduleExactAlarms(alarmManager)) {
+            AlarmManagerCompat.setExactAndAllowWhileIdle(
+                alarmManager,
+                AlarmManager.ELAPSED_REALTIME_WAKEUP,
+                triggerTime,
+                pendingIntent,
+            )
+        } else {
+            alarmManager.set(AlarmManager.ELAPSED_REALTIME_WAKEUP, triggerTime, pendingIntent)
+        }
```

Comment:
This is not equivalent to the previous behavior. It's an error to call `AndroidAlarmManager.setAlarm()` without the permission to schedule alarms. So we want `AlarmManager` to throw an exception when that happens, not fall back to `AlarmManager.set()`.

```suggestion
        AlarmManagerCompat.setExactAndAllowWhileIdle(
            alarmManager,
            AlarmManager.ELAPSED_REALTIME_WAKEUP,
            triggerTime,
            pendingIntent,
        )
```